source/common/formatter/header_formatter.h

```diff
@@ -9,6 +9,8 @@
 #include "absl/strings/string_view.h"
 
 namespace Envoy {
+
+// TODO(auni53): Fix the namespace and rename this class.
```

Comment:
I don't think the class needs to be renamed IMHO but can you fix the namespace in this PR?

Reply:
It'll have to be renamed, as there's already a Formatter::HeaderFormatter in substitution_formatter.h. I figured if there's code reuse I should figure that out separately, since I don't really know anything about the former accesslog formatter or where use cases intersect.

Reply:
Yes let's not figure out the reuse right now but let's do some type of rename. I would probably actually rename the one in substitution_formatter.h to something like HeaderFormatterImpl or SubsitituionHeaderFormatter with a TODO/comment about code sharing with this thing.

Reply:
That's fine, but if I'm changing the namespaces I might as well change the names. I'm going to rename one to SubstitutionHeaderFormatter and the other to CustomHeaderFormatter so that whoever eventually tries to refactor or merge these doesn't get confused about "which is the true HeaderFormatter"/

Reply:
If we are really going to bikeshed about names, I don't think CustomHeaderFormatter really makes any sense.

If anything, I would call this one SubstitutionHeaderFormatter because that is what it is. Rename the other one anything you want.

Reply:
Actually, yes, please do that, and call the other one `SubsitutionFromHeaderFormatter`. Thank you!

Reply:
I lost track of what you're referring to. CustomHeaderFormatter is because this is formatting/substituting header values, based on an input custom header. Which are you saying to call SubstitutionFromHeaderFormatter?

Reply:
Let's do this:
- Call this class that you have moved `SubstitutionHeaderFormatter` because it is formatting s header based on substitutions.
- Call the inner class in the other file `SubsitutionFromHeaderFormatter` because this is what it is doing, creating a substitution from a header.

Reply:
Latter makes sense to me, I opted to hold off on the former as I have some thoughts*. Made the namespace change and pushed.

* The custom header feature isn't really formatting the headers themselves, they're formatting the header values. ValueFormatter? HeaderValueFormatter? In any case I'd rather leave as is or for another PR.